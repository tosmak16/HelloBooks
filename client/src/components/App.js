import React from 'react';
import PropTypes from 'prop-types';

import NavigationBar from './NavigationBar';


/**
 * 
 * 
 * @class App
 * @extends {React.Component}
 */
class App extends React.Component {
  /**
   * 
   * 
   * @returns 
   * @memberof App
   */
  render() {
    return (

      <div>

        <NavigationBar />
        {this.props.children}

      </div>


    );
  }
}

App.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,

};

export default App;
