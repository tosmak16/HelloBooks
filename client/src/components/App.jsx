import React from 'react';
import PropTypes from 'prop-types';

import NavigationBar from './NavigationBar';


/**
 * @function App
 * @param {object} props
 * @returns {views} containing Nav bar and some other component properties
 */
function App(props) {
  /**
   * @returns
   * @memberof App
   */
  return (
    <div>
      <NavigationBar />
      {props.children}
    </div>
  );
}
App.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,

};

export default App;
