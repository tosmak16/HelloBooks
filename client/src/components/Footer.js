import React from 'react';
import { Link } from 'react-router';

/**
 * 
 * 
 * @class Footer
 * @extends {React.Component}
 */
class Footer extends React.Component {
  /**
   * 
   * 
   * @returns 
   * @memberof Footer
   */
  render() {
    return (
      <footer className="footer-distributed ">

        <div className="row">
          <div className=" col l12 col m12 col s12">

            <div className="footer-left col l4 col m4 col s12">

              <h3>Hello<span>Books</span></h3>

              <p className="footer-links">
                <Link to={'http//:'}>Home</Link>
                <Link to={'http//:'}>Blog</Link>
                <Link to={'http//:'}>Pricing</Link>
                <Link to={'http//:'}>About</Link>
                <Link to={'http//:'}>Faq</Link>
                <Link to={'http//:'}>Contact</Link>
              </p>

              <p className="footer-company-name">Tosmak &copy; 2017</p>
            </div>

            <div className="footer-center col l4 col m4 col s12">

              <div>
                <i className="fa fa-map-marker" />
                <p><span>21 Ori-oke Street</span> Lagos, Nigeria</p>
              </div>

              <div>
                <i className="fa fa-phone" />
                <p>08127112053</p>
              </div>

              <div>
                <i className="fa fa-envelope" />
                <p><Link href="tosmak16@gmail.com">tosmak16@gmail.com</Link></p>
              </div>

            </div>

            <div className="footer-right col l4 col m14 col s12">

              <div className="footer-company-about">
                <span>
                  About Hello Book
                </span>
                <p>
                  it is a simple application that helps manage a library and
                  its processes like stocking,tracking and renting books
                </p>
              </div>

              <div className="footer-icons">

                <Link to={'http//:'}><i className="fa fa-facebook" /></Link>
                <Link to={'http//:'}><i className="fa fa-twitter" /></Link>
                <Link to={'http//:'}><i className="fa fa-linkedin" /></Link>
                <Link to={'http//:'}><i className="fa fa-github" /></Link>

              </div>

            </div>
          </div>
        </div>

      </footer>
    );
  }
}

export default Footer;
