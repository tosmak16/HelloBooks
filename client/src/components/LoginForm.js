import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Icon, Input, Row } from 'react-materialize';

import userSignin from '../actions/loginAction';
import getbooks from '../actions/getBooks';
import log from '../../reducers/log';

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ errors: '' });
    this.props.userSignin(this.state);
    this.props.getbooks(true);
  }

  render() {
    const { error } = this.props.log;
    return (
      <form onSubmit={ this.handleSubmit } className="form-signin" action="books.html" >
        <h4 className="form-signin-heading">Please log in</h4>
        {error && <p style={{ color: 'red' }} className="help-block">*{error}*</p>}
        <Row>
          <Input
            s={ 12 } label="Username"
            type="text" value={ this.state.username } name="username" id="inputUsernamelog" className="form-control validate"
            placeholder="Username"
            onChange={ this.handleChange }
            required
            autoFocus
          />
        </Row>

        <Row>
          <Input
            s={ 12 } label="Password"
            type="password"
            id="inputPassword"
            className="form-control validate"
            placeholder="Password" required
            autoFocus
            name="password"
            value={ this.state.password }
            onChange={ this.handleChange }
          />
        </Row>

        <div className="row">
          <div>
            <input
              defaultChecked="checked" name="group1" type="checkbox" value="" label="Remember me?"
              className="filled-in checkbox-orange" id="filled-in-box"
            />
            <label htmlFor="filled-in-box">Remember me?</label>
          </div>
        </div>
        <button id="loginbtn" className="btn btn-lg btn-primary  btn-block" type="submit">Sign in</button>
        <p>
          <label>
            <a href=""> Forgot Password ? </a>
          </label>
        </p>

        <a href="/signup" id="createbtn" className="" > <b>Create account</b></a>

      </form >
    );
  }
}

LoginForm.propTypes = {
  getbooks: PropTypes.func.isRequired,
  log: PropTypes.object.isRequired,
  userSignin: PropTypes.func.isRequired,

};

const mapStateToProps = state => ({
  log: state.log[0]
});

export default connect(mapStateToProps, { userSignin, getbooks })(LoginForm);
