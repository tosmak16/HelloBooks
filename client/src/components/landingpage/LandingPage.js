import React from 'react';
import { Link } from 'react-router';


class LandingPage extends React.Component {
  render() {
    return (
      <div className="welcome">
        <span id="splash-overlay" className="splash" />
        <span id="welcome" className="z-depth-4" />
        <main className="valign-wrapper">
          <span className="container grey-text text-lighten-1 ">

            <p style={{ fontSize: '35px', color: '#3e1111' }} className="flow-text">Welcome to</p>
            <h1 style={{ color: 'rgba(142, 15, 15, 0.74)' }} className="title text-lighten-3">Hello Books</h1>

            <blockquote style={{ color: 'rgba(142, 15, 15, 0.74)' }} className="flow-text">it's a simple application that helps manage a library and its processes like stocking,  tracking and renting books</blockquote>

            <div style={{ boxShadow: '1px 1px 50px 20px rgba(70, 34, 9, 0.35)' }} className="center-align">

              <Link
                className="btn"
                to={ '/login' }
                data-activates="exams"
                style={{ backgroundColor: '#3e1111' }}
              >Sign In</Link>

              <Link
                style={{ backgroundColor: '#93776d' }}
                className="btn"
                to={ '/signup' }
                data-activates="study"
              >Sign Up</Link>
            </div>

          </span>
        </main>
      </div>
    );
  }
}


export default LandingPage;
