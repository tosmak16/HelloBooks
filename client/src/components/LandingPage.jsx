import React from 'react';
import { Link } from 'react-router';
import { Footer } from '../components/Footer';

export const LandingPage = () => {
  document.title = 'Home Page';
  return (
    <div>
      <div className="con">
        <div className="row">
          <div className="col-lg-12">
            <div className="content">
              <h1>Hello Books</h1>
              <h3>“A mind needs books as a sword needs a whetstone.” </h3>
              <hr className="hr-landing" />
              <Link to="/signup" className="btn btn-default btn-lg landing-button">
                Sign up</Link>
            </div>
          </div>
        </div>
      </div>
      <div className="footer-set">
        <Footer />
      </div>
    </div>
  );
};
export default LandingPage;
