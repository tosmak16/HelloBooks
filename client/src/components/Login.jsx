import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LogininForm from './LoginForm';
import { Footer } from './Footer';
import userSignin from '../actions/login';
import { getBooks } from '../actions/getBooks';
import { googleAuthSignIn } from '../actions/googleAuthSignIn';
import { resetUserPassword } from '../actions/resetUserPassword';

/**
 * @description connnected login page
 * 
 * @param {object} props
 * 
 * @export Login
 * 
 * @function Login
 * 
 * @returns {views} login form
 */
export const Login = props => (
  <div>
    <div id="logininForm" className="row">
      <LogininForm
        getbooks={props.getBooks}
        userSignin={props.userSignin}
        googleAuthSignIn={props.googleAuthSignIn}
        login={props.login}
        resetUserPassword={props.resetUserPassword}
        resetPasswordStatus={props.resetPasswordStatus}
      />
    </div>
    <Footer />
  </div>
);
Login.propTypes = {
  getBooks: PropTypes.func.isRequired,
  login: PropTypes.objectOf(PropTypes.any).isRequired,
  userSignin: PropTypes.func.isRequired,
  googleAuthSignIn: PropTypes.func.isRequired,
  resetUserPassword: PropTypes.func.isRequired,
  resetPasswordStatus: PropTypes.string.isRequired
};
/**
 *@function mapStateToProps

 * @param {object} state
 * 
 * @returns {object} login
 */
function mapStateToProps(state) {
  return {
    login: state.login[0],
    resetPasswordStatus: state.resetPassword[0].data
  };
}
export default connect((mapStateToProps),
  {
    userSignin,
    getBooks,
    googleAuthSignIn,
    resetUserPassword
  })(Login);
