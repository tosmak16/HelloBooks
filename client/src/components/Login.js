import React from 'react';
import LoginForm from './LoginForm';
import Footer from './Footer';
// import '../../public/scss/index.scss';
import '../../public/scss/materialize.scss';
import FlashMessagesList from './flash/FlashMessagesList';


class Login extends React.Component {
  render() {
    return (
      <div>
        <FlashMessagesList />
        <div id="loginForm" className="row">
          <LoginForm />
        </div>
        <Footer />
      </div>
    );
  }
}

export default Login;
