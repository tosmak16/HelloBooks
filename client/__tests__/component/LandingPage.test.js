import React from 'react';
import render from 'react-test-renderer';
import expect from 'expect';
import { LandingPage } from '../../src/components/LandingPage';


it('should render and take snapshot of booksPage', () => {
  const tree = render.create(<LandingPage />);
  expect(tree).toMatchSnapshot();
});
