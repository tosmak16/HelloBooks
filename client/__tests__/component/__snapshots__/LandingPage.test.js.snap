// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and take snapshot of booksPage 1`] = `
<div>
  <div
    className="con"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="content"
        >
          <h1>
            Hello Books
          </h1>
          <h3>
            “A mind needs books as a sword needs a whetstone.” 
          </h3>
          <hr
            className="hr-landing"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer-set"
  >
    <footer
      className="footer-distributed "
    >
      <div
        className="row"
      >
        <div
          className=" col l12 col m12 col s12"
        >
          <div
            className="footer-left col l4 col m4 col s12"
          >
            <h3>
              Hello
              <span>
                Books
              </span>
            </h3>
            <p
              className="footer-company-name"
            >
              Tosmak © 2017
            </p>
          </div>
          <div
            className="footer-center col l4 col m4 col s12"
          >
            <div>
              <i
                className="fa fa-map-marker"
              />
              <p>
                <span>
                  21 Ori-oke Street
                </span>
                 Lagos, Nigeria
              </p>
            </div>
            <div>
              <i
                className="fa fa-phone"
              />
              <p>
                08127112053
              </p>
            </div>
            <div>
              <i
                className="fa fa-envelope"
              />
              <p />
            </div>
          </div>
          <div
            className="footer-right col l4 col m14 col s12"
          >
            <div
              className="footer-company-about"
            >
              <span>
                About Hello Book
              </span>
              <p>
                it is a simple application that helps manage a library and its processes like stocking,tracking and renting books
              </p>
            </div>
            <div
              className="footer-icons"
            />
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
